### PR TITLE
GEODE-5578 Create small docker image for meta-pipeline jobs

### PR DESCRIPTION
**Pull request by rhoughton-pivot**
Meta pipeline hosts a job that creates an image used in the other jobs.
Resolves chicken/egg issue with create-images-pipeline and meta itself.
Fix version of concourse-pipeline resource type. Improve pause behavior for meta pipeline

Co-authored-by: Robert Houghton <rhoughton@pivotal.io>
Co-authored-by: Finn Southerland <fsoutherland@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by rhoughton-pivot** [comment]
@smgoller @metatype Can you review this? I want your take on the new image contents/dockerfile as well

**Comment by smgoller** [comment]
Seems legit.

File: ci/pipelines/deploy_meta.sh [hunk 1]
```diff
@@ -41,19 +42,30 @@ fly -t ${TARGET} set-pipeline \
...
+    fly -t ${TARGET} pause-job \
+        -j ${META_PIPELINE}/${job}-pipeline
+  done
+
+  fly -t ${TARGET} trigger-job \
+      -j ${META_PIPELINE}/build-meta-mini-docker-image
+  fly -t ${TARGET} unpause-pipeline \
+      -p ${META_PIPELINE}
   set +x
 fi
```
Comment (pivotal-jbarrett) [inline ci/pipelines/deploy_meta.sh]: Missing line feed here.

File: ci/pipelines/meta.yml [hunk 2]
```diff
@@ -229,3 +257,15 @@ jobs:
...
+  public: ((!public-pipelines))
+  serial: true
+  plan:
+  - aggregate:
+    - get: meta-mini-dockerfile
+      trigger: true
+  - put: meta-mini-image
+    params:
+      build: meta-mini-dockerfile/ci/images/meta-mini
+      tag_as_latest: true
```
Comment (pivotal-jbarrett) [inline ci/pipelines/meta.yml]: Missing line feed here.

File: ci/images/meta-mini/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+FROM alpine:3.7
+
+RUN apk --no-cache add \
+      bash \
+      curl \
+      jq \
+      python3 \
+  && curl -L -o /usr/local/bin/spruce https://github.com/geofffranks/spruce/releases/download/$(curl --silent "https://api.github.com/repos/geofffranks/spruce/releases/latest" | jq -r .tag_name)/spruce-linux-amd64 \
+  && curl -L -o /usr/local/bin/fly "https://concourse.apachegeode-ci.info/api/v1/cli?arch=amd64&platform=linux" \
+  && chmod +x /usr/local/bin/* \
```
Comment (pivotal-jbarrett) [inline ci/images/meta-mini/Dockerfile]: Where is the line being continued?

**Pull request merged**